### PR TITLE
[NEW] Support 'Accept-Language' HTTP header to localize error msgs

### LINKED ISSUE #10817: [API] Localize our REST API messages based on HTTP header
Main problem is our error messages that always come in the english language.

Here is the standard we should be following:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language

### PR DESCRIPTION
**Pull request by cardoso**
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language

Refs #10817

### CODE REVIEW

**Comment by tassoevan** [comment]
Will the `TAPi18n.__(key, options, lang_tag=null)` function ignore a random `lang_tag` value? I'm worried about `Accept-Language` be exploited somehow.

File: packages/rocketchat-api/server/api.js [hunk 1]
```diff
@@ -170,6 +170,12 @@ class API extends Restivus {
 
 						result = result || RocketChat.API.v1.success();
 
+						const lang = this.request.headers['accept-language'];
```
Comment (ggazzo) [inline packages/rocketchat-api/server/api.js]:
```javascript
	failure(result, errorType) {
		if (_.isObject(result)) {
			result.success = false;
		} else {
			result = {
				success: false,
				error: result
			};

			if (errorType) {
				result.errorType = errorType;
			}
		}

		result = {
			statusCode: 400,
			body: result
		};

		logger.debug('Failure', result);

		return result;
	}
```

**Comment by ggazzo** [comment]
@sampaiodiego

**Comment by cardoso** [comment]
@ggazzo this PR adds the feature. New PRs should support it.

As far as I tested it, errors without localization keys still work, they just won't be localized.

If it can be made a priority, which I think it should for the sake of our mobile users, we can assign someone to go back and add localization to all errors that don't have it.

**Comment by engelgabriel** [comment]
I think this PR is a step in the right direction, but it does not close #10817 

Regarding the translation of error messages... I have some points for you to consider:
1. We will have to translate many other messages on the client side anyway (because they are sent to all member of a room) like USER_JOINDED, etc... so building the capability of loading the translation files from the server is inevitable. 
2. Error codes are also very fundamental to ANY good practice and we MUST make sure we use it in *ALL* erros, and not just a few.
3. Having an error payload with the variables also helps on the debug if they can be shown on user request.
4. We should build a fallback approach, where we can always have an english version if the translation fails on each level.

**Pull request closed without merging**